content/articles/2018-10-23-dplyr-0-8-0-release-candidate.Rmd

```diff
@@ -0,0 +1,176 @@
+---
+title: dplyr 0.8.0 release candidate
+author: Romain François
+date: '2018-10-23'
+slug: dplyr-0-8-0-release-candidate
+description: > 
+  dplyr 0.8.0 release candidate
+categories:
+  - package
+tags:
+  - dplyr
+  - tidyverse
+---
+
+```{r, echo = FALSE, message = FALSE}
+knitr::opts_chunk$set(collapse = T, comment = "#>")
+options(tibble.print_min = 4L, tibble.print_max = 4L)
+library(dplyr)
+```
+
+# Tidy grouping structure
+
+Previous versions of `dplyr` used a messy set of attributes in grouped
+tibbles to keep track of the groups and their indices. This has been 
+re-organized into a tibble that can be accessed with the new 
+`group_data()` function. 
+
+```{r}
+iris %>% 
+  group_by(Species) %>% 
+  group_data()
+```
+
+The first columns of that tibble describe the groups in terms of the 
+grouping variables, and the last column (always called `.rows`)
+is a list of integer vectors identifying the (one-based) indices of 
+each group. 
+
+The related function `group_rows()` gives just that last column. 
+
+```{r}
+iris %>% 
+  group_by(Species) %>% 
+  group_rows()
+```
+
+# New grouping algorithm
+
+The algorithm behind `group_by()` has been redesigned to better respect factors. 
+In previous versions of `dplyr` groups were created to only match the observed data,
+which was potentially losing groups when factors are involved. 
+
+The new algorithm respects the levels of the factors that are involved in the 
+grouping, by creating groups even if there is no data in that group. 
+
+```{r}
+df <- tibble(
+  x = c(1,1,2,2), 
+  f = factor(c("a", "b", "a", "b"), levels = c("a", "b", "c"))
+)
+df %>% 
```

Comment:
I think it would be better to illustrate these with `count()`, and maybe a single example with `summarise(mean(x))` (to reinforce what happens with zero-row inputs to summary functions)